worlds/ahit/Types.py

```diff
@@ -0,0 +1,87 @@
+from enum import IntEnum, IntFlag
+from typing import NamedTuple, Optional, List
+from BaseClasses import Location, Item, ItemClassification
+
+
+class HatInTimeLocation(Location):
+    game = "A Hat in Time"
+
+
+class HatInTimeItem(Item):
+    game = "A Hat in Time"
+
+
+class HatType(IntEnum):
+    NONE = -1
+    SPRINT = 0
+    BREWING = 1
+    ICE = 2
+    DWELLER = 3
+    TIME_STOP = 4
+
+
+class HitType(IntEnum):
+    none = 0
+    umbrella = 1
+    umbrella_or_brewing = 2
+    dweller_bell = 3
+
+
+class HatDLC(IntFlag):
+    none = 0b000
+    dlc1 = 0b001
+    dlc2 = 0b010
+    death_wish = 0b100
+    dlc1_dw = 0b101
+    dlc2_dw = 0b110
+
+
+class ChapterIndex(IntEnum):
+    SPACESHIP = 0
+    MAFIA = 1
+    BIRDS = 2
+    SUBCON = 3
+    ALPINE = 4
+    FINALE = 5
+    CRUISE = 6
+    METRO = 7
+
+
+class Difficulty(IntEnum):
+    NORMAL = -1
+    MODERATE = 0
+    HARD = 1
+    EXPERT = 2
+
+
+class LocData(NamedTuple):
+    id: Optional[int] = 0
+    region: Optional[str] = ""
+    required_hats: Optional[List[HatType]] = [HatType.NONE]
```

Comment:
if you ever modify this list, this base one will be shared between all of the locations that don't have a non default list.

Reply:
I figured that `Optional` was necessary for optional variables, and this default list was never meant to be changed. But since I can see that it's not actually necessary, I removed `Optional` from the class entirely and changed this to an empty list.